### DISCUSSION TITLE
mbuffer: "operation not supported" message and not sure how to troubleshoot

### DISCUSSION THREAD

**Discussion by emory** [oetiker/znapzend#648] [Q&A]
hello,

i'm using mbuffer on macOS in my znapzend configuration via:

     mbuffer=/usr/local/bin/mbuffer
     mbuffer_size=1G

and when i'm running `znapzend --debug` i see error messages for my configurations using mbuffer, ==redaction via `…`==:

```
# zfs send -I cascade/…/notes@2023-07-26-113000 cascade/…/notes@2023-07-26-114500|/usr/local/bin/mbuffer -q -s 256k -W 600 -m 1G|zfs recv -F succor/localbackup/iow/icon/cascade/…/notes
mbuffer: warning: unable to sync <stdout>: Operation not supported
mbuffer: warning: unable to sync <stdout>: Operation not supported
# zfs list -H -o name -t snapshot cascade/av@2023-07-26-114500
```
my `mbuffer` binary i built myself per the instructions in a GitHub issue: https://github.com/ilovezfs/mbuffer-osx/issues/1

the version of my local `mbuffer` is `20141227`, i've built it on Apple Silicon and intel and i use openzfs for macOS on two workstations. i use `znapzend` for local pool-to-pool backups and plan to begin using it for host-to-host, i hadn't noticed the error until recently. i haven't setup any launchd jobs or anything yet, i foreground and `--debug` in a tmux pane while i get familiar still.

**Comment by lundman** [comment]
comes from
`err = fsync(fd);`
spit out by
`warningmsg("unable to sync %s: %s\n",d->arg,strerror(errno));`

but you are already in `terminateSender()` at this point. So either it worked, and you are getting noise at the end, or something else went wrong.

[a comment by an automated account is not shown]

**Comment by xenophon61** [comment]
Sorry for the noob question - is it still possible to compile mbuffer for MacOS?

 If so, how? (the 2018 [instructions](https://github.com/ilovezfs/mbuffer-osx/issues/1) by ilovezfs don't appear to work).

TIA

**Reply by 2600box** [reply]
> comes from `err = fsync(fd);` spit out by `warningmsg("unable to sync %s: %s\n",d->arg,strerror(errno));`
> 
> but you are already in `terminateSender()` at this point. So either it worked, and you are getting noise at the end, or something else went wrong.

@lundman My testing with [zfs-check](https://github.com/psy0rz/zfs_autobackup/wiki/zfs-check) shows the correct hashes, but it is a bit concerning. Would you be willing to take a look at resolving the "noise" issue?


> Sorry for the noob question - is it still possible to compile mbuffer for MacOS?
> 
> If so, how? (the 2018 [instructions](https://github.com/ilovezfs/mbuffer-osx/issues/1) by ilovezfs don't appear to work).
> 
> TIA

@xenophon61 You have almost certainly succeeded or moved on by now, but just in case I published my build for Apple Silicon here: [https://github.com/2600box/mbuffer-osx/releases/tag/AppleSilicon](https://github.com/2600box/mbuffer-osx/releases/tag/AppleSilicon)

Be warned I have no idea what I am doing, but it seems to work fine, aside from the error reported above, `mbuffer: warning: unable to sync <stdout>: Operation not supported`
